week-3/Homework/mandatory/7-password-validator.js

```diff
@@ -22,8 +22,25 @@ PasswordValidationResult=  [false, false, false, false, true]
 
 */
 
-function validatePasswords(passwords) {
+function checkPassword(password,index,arr) {
+  let passUppercase = /[A-Z]/;
+  let passLowerercase = /[a-z]/;
+  let passNumber = /[0-9]/;
+  let passSymbols = /[!#$%.]/;
 
+  return (
+    password.length >= 5 &&
+    passUppercase.test(password) &&
+    passLowerercase.test(password) &&
+    passNumber.test(password) &&
+    passSymbols.test(password)&&
+    arr.indexOf(password) === index
+  );
+}
+
+function validatePasswords(passWordList) {
+  let newPass = passWordList.map(checkPassword);
+  return newPass;
```

Comment:
Here's another opportunity to inline the result into the `return` statement. But looking really great!

Reply:
`function validatePasswords(passWordList) {
  return  passWordList.map(checkPassword);
  
}`

Reply:
`function validatePasswords(passWordList) {
return passWordList.map(checkPassword);}`

Reply:
🚀